src/Microsoft.DocAsCode.Metadata.ManagedReference.Common/ExtractMetadata/TocNamespaceStyle.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Metadata.ManagedReference
+{
+    public enum TocNamespaceStyle
+    {
+        /// <summary>
+        /// Renders the namespaces as a single flat list
+        /// </summary>
+        Flattened,
+
+        /// <summary>
+        /// Renders the namespaces in a nested tree form
+        /// </summary>
+        Nested,
+
+        /// <summary>
+        /// Renders the namespaces in a nested tree form, compacting namespaces to only use the lowest level
+        /// </summary>
+        CompactNested,
```

Comment:
I feel a bit hesitated to expose both `nested` and `compactNested` in the first release of the feature. We probably want to pick one recommended nesting style. My intuition is to show only the lowest level as it is more condensed, it matches nested namespace declaration syntax in C# and it fits naturally with the treeview. So consider remove the `CompactNested` option and adjust the behavior of `Nested` to match what `CompactNested` is describing here.